servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -488,4 +490,16 @@ private static int verifyPortRange(final int port) {
         }
         return port;
     }
+
+    private static void setRequestAuthority(HttpRequestMetaData metaData) {
+        if (metaData.context().containsKey(AUTHORITY_KEY) || metaData.host() == null) {
+            return;
+        }
+        // the host from the metaData should already be sanitized.
+        CharSequence authority = metaData.host();
+        if (metaData.port() >= 0) {
+            authority = authority + ":" + metaData.port();
```

Comment:
is there a chance this could be ipv6 address, and if so does it need to be enclosed in brackets: `[]`

Reply:
It _should_ already be properly bracketed and is in the `DefaultHttpRequestMetadata`. That doesn't preclude broken alternative `HttpRequestMetaData` implementations but they could do basically anything they want so it's not easy to mitigate other than making another trip through `Uri3986`.
```
// AbstractHttpRequestMetaDataTest.java
@Test
void testParseHttpsUriAbsoluteFormWithIpV6WithPort() {
    createFixture("https://[1:2:3::5]:8080/path");
    assertEquals("[1:2:3::5]:8080", fixture.host() + ":" + fixture.port());
    assertEffectiveHostAndPort("1:2:3::5", 8080);

    createFixture("https://[1:2:3::5]/path");
    assertEquals("[1:2:3::5]", fixture.host());
    assertEffectiveHostAndPort("1:2:3::5");
// Passes.
}
```